bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,270 @@
+                } else
+                    logger.debug("initNetwork alreadys started");
+                updateStatus();
+                return;
+            }
+            logger.debug("State should be ok !!!!!!");
+        } catch (CGateException e) {
+            logger.warn("Cannot load C-Bus network {}", networkID, e);
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        updateStatus();
+    }
+
+    private void checkNetworkOnline() {
+        Network network = getNetwork();
+        try {
+            if (network != null && network.isOnline()) {
+                logger.debug("Network is online");
+                ScheduledFuture<?> initNetwork = this.initNetwork;
+                if (initNetwork != null)
+                    initNetwork.cancel(false);
+                updateStatus();
+            } else {
+                ThingStatus lastStatus = getThing().getStatus();
+                logger.debug("Network still not online {}", lastStatus);
+                updateStatus();
+            }
+        } catch (CGateException e) {
+            logger.warn("Cannot check if network is online {} ", network.getNetworkID());
+            updateStatus();
+        }
+    }
+
+    private void updateStatus() {
+        logger.debug("updateStatus");
+        ThingStatus lastStatus = getThing().getStatus();
+        Network network = getNetwork();
+        CBusCGateHandler cbusCGateHandler = getCBusCGateHandler();
+        try {
+            if (cbusCGateHandler == null || !cbusCGateHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "CGate connection offline");
+            } else if (network == null) {
+                logger.debug("No network - set configuration error");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            } else if (network.isOnline()) {
+                logger.debug("Network is online");
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.debug("Network is offline");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Network is not reporting online");
+            }
+        } catch (CGateException e) {
+            logger.warn("Problem checking network state for network {}", network.getNetworkID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        if (!getThing().getStatus().equals(lastStatus)) {
+            ScheduledFuture<?> networkSync = this.networkSync;
+            if (lastStatus == ThingStatus.OFFLINE) {
+                if (networkSync == null || networkSync.isCancelled())
+                    this.networkSync = scheduler.scheduleWithFixedDelay(() -> {
+                        doNetworkSync();
+                    }, (int) (10 * Math.random()),
```

Comment:
use Random.nextInt instead

Reply:
I dont see why its random. We start it once so i've made i a fixed delay.